iis-common/src/main/java/eu/dnetlib/iis/common/pig/udfs/DeduplicateIdsWithConfidence.java

```diff
@@ -13,55 +13,76 @@
 import org.apache.pig.impl.logicalLayer.schema.Schema;
 
 /**
- * Deduplicates bag of tuples where tuple[0] is identifier and tuple[1] is confidence level.
- * Highest confidence level is picked when identifier duplicate is found.
- * Identifiers are sorted lexicographically.
+ * Deduplicates bag of tuples where tuple[0] is identifier and tuple[confidenceLevelPosition] is confidence level. 
+ * Highest confidence level is picked when identifier duplicate is found. Identifiers are sorted lexicographically.
  *
  * @author mhorst
  */
 public class DeduplicateIdsWithConfidence extends EvalFunc<DataBag> {
 
+    /**
+     * Confidence level position in tuple. First element is indexed with 0.
+     */
+    private final int confidenceLevelPosition;
+    
+    //------------------------ CONSTRUCTORS --------------------------
+
+    public DeduplicateIdsWithConfidence() {
+        this(1);
+    }
+    
+    public DeduplicateIdsWithConfidence(int confidenceLevelPosition) {
+        this.confidenceLevelPosition = confidenceLevelPosition;
+    }
+    
+    public DeduplicateIdsWithConfidence(String confidenceLevelPosition) {
```

Comment:
It's very strange. If you pass a string that is not an integer, you'll get exception, so why do you allow it?